edstem-api/edstem_api/api.py

```diff
@@ -0,0 +1,126 @@
+
+import json as _json
+import typing as _typing
+
+import requests as _requests
+import six as _six
+import wsse.client.requests.auth as _wsse_auth
+
+import exceptions
+
+API_BASE_URL = "https://us.edstem.org/api/"
+
+last_token = None
+
+config = {
+    "username": "netid@cornell.edu",
+    "password": "password",
+}
+
+
+def get_auth_token(username=None, password=None):
+    # type: (_typing.Optional[str], _typing.Optional[str]) -> _typing.Optional[str]
+    """
+    Obtain an authentication JWT for the Ed platform, given a valid username and
+    password, provided in the configuration file.
+    :param username: (Optionally) override configuration file username
+    :param password: (Optionally) override configuration file password
+    :return: A JWT to access the Ed platform's API.
+    """
+    global last_token
+
+    url = _six.moves.urllib.parse.urljoin(
+            base=API_BASE_URL,
+            url="token",
+        )
+
+    try:
+        res = _requests.post(
+            url=url,
+            headers={
+                "Content-Type": "application/json; charset=utf-8",
+            },
+            data=_json.dumps({
+                "login": username or config["username"],
+                "password": password or config["password"]
+            })
+        )
+
+    except _requests.exceptions.RequestException:
+        raise
+
+    if res.status_code != 200:
+        return None
+
+    try:
+        data = res.json()
+    except ValueError:
+        raise
+
+    token = data.get("token", None)
+    last_token = token
+
+    return token
+
+
+def request(endpoint=None, url=None, data=None, json=None, **kwargs):
+    # type: (_typing.Optional[str], _typing.Optional[str], _typing.Optional[dict], _typing.Dict) -> _typing.DefaultDict
+    """
+    Make a request directly to the Ed platform's API.
+    """
```

Comment:
This is something I will probably need to do as a whole but we should go through and document things like we documented [this function](https://github.com/oscarso2000/MyCourseIndex/pull/182/files#diff-55a2e1060a9cc44a865f70144bfde3ae33d01a3fee713f09d4b2cd4432405eb5R21).